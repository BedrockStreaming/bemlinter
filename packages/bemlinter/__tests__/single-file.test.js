const {lint, format} = require('../src/bemlinter.js');

const snapLintOutput = (fileName, done, options = {}) => {
  lint(`${__dirname}/styles/${fileName}`, options)
    .then(lintResult => format(lintResult, false))
    .then(output => {
      expect(output).toMatchSnapshot();
    })
    .catch(console.error)
    .then(done)
  ;
};

describe('Bemlinter of alright.scss', () => {
  it('should lint without error', done => snapLintOutput(`alright.scss`, done));
});

describe('Bemlinter of ModuleAlright.scss', () => {
  it('should lint without error', done => snapLintOutput(`ModuleAlright.scss`, done, {
    filePattern: '(?:Module)?([^.]*)\.scss'
  }));
});

describe('Bemlinter of warning.scss', () => {
  it('should warn the use of an external block', done => snapLintOutput(`warning.scss`, done));
});

describe('Bemlinter of prefix.scss', () => {
  it('should lint with a missing prefix error', done => snapLintOutput(`prefix.scss`, done, {classPrefix: 'c-'}));
});

describe('Bemlinter of leak.scss', () => {
  it('should lint with a leak error', done => snapLintOutput(`leak.scss`, done));
});

describe('Bemlinter of leak.css', () => {
  it('should lint with a leak error', done => snapLintOutput(`leak.css`, done));
});

describe('Bemlinter of mixins-content.scss', () => {
  it('should lint with a leak error', done => snapLintOutput(`mixins-content.scss`, done));
});

describe('Bemlinter of syntax.scss', () => {
  it('should lint with 6 different syntax error', done => snapLintOutput(`syntax.scss`, done));

  it('should lint without a lower case error', done => snapLintOutput(`syntax.scss`, done, {checkLowerCase: false}));
});

describe('Bemlinter of class-concat.scss', () => {
  it('should detect class concatenation', done => snapLintOutput(`class-concat.scss`, done));
});

describe('Bemlinter of unreadable.scss', () => {
  it('should log a block as unreadable', done => snapLintOutput(`unreadable.scss`, done));
});
