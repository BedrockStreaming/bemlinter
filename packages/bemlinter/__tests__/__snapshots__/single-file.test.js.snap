exports[`Bemlinter of ModuleAlright.scss should lint without error 1`] = `"OK: bemlinter has validated 1 block."`;

exports[`Bemlinter of alright.scss should lint without error 1`] = `"OK: bemlinter has validated 1 block."`;

exports[`Bemlinter of class-concat.scss should detect class concatenation 1`] = `
"  ✗ class-concat
[class-concat.scss:23] Error: \"&__close\" should not concatenate classes.
[class-concat.scss:31] Error: \"&--big\" should not concatenate classes.

FAIL: bemlinter has detected 2 errors on 1 block."
`;

exports[`Bemlinter of leak.css should lint with a leak error 1`] = `
"  ✗ leak
[leak.css:11] Error: \".external-block\" is incoherent with the file name.

FAIL: bemlinter has detected 1 error on 1 block."
`;

exports[`Bemlinter of leak.scss should lint with a leak error 1`] = `
"  ✗ leak
[leak.scss:10] Error: \".external-block\" is incoherent with the file name.

FAIL: bemlinter has detected 1 error on 1 block."
`;

exports[`Bemlinter of mixins-content.scss should lint with a leak error 1`] = `
"  ✗ mixins-content
[mixins-content.scss:7] Error: \".icon\" is incoherent with the file name.

FAIL: bemlinter has detected 1 error on 1 block."
`;

exports[`Bemlinter of prefix.scss should lint with a missing prefix error 1`] = `
"  ✗ prefix
[prefix.scss:21] Error: \".prefix--big\" should have a block prefix.

FAIL: bemlinter has detected 1 error on 1 block."
`;

exports[`Bemlinter of syntax.scss should lint with 6 different syntax error 1`] = `
"  ✗ syntax
[syntax.scss:21] Error: \".syntax---big\" modifier should have only 2 dashes.
[syntax.scss:26] Error: \".syntax--big--light\" should have a single modifier.
[syntax.scss:32] Error: \".syntax__Close\" should be in lower case.
[syntax.scss:40] Error: \".syntax--big__close\" represents an element of a modifier, it should be cut in 2 classes.
[syntax.scss:49] Error: \".syntax__header__logo\" should have a single depth of element.
[syntax.scss:53] Error: \".syntax___footer\" element should have only 2 underscores.

FAIL: bemlinter has detected 6 errors on 1 block."
`;

exports[`Bemlinter of syntax.scss should lint without a lower case error 1`] = `
"  ✗ syntax
[syntax.scss:21] Error: \".syntax---big\" modifier should have only 2 dashes.
[syntax.scss:26] Error: \".syntax--big--light\" should have a single modifier.
[syntax.scss:40] Error: \".syntax--big__close\" represents an element of a modifier, it should be cut in 2 classes.
[syntax.scss:49] Error: \".syntax__header__logo\" should have a single depth of element.
[syntax.scss:53] Error: \".syntax___footer\" element should have only 2 underscores.

FAIL: bemlinter has detected 5 errors on 1 block."
`;

exports[`Bemlinter of unreadable.scss should log a block as unreadable 1`] = `
"  ✗ unreadable
[unreadable.scss] Error: Can\'t handle character: \"Û\" (1:0)

FAIL: bemlinter has detected 1 error on 1 block."
`;

exports[`Bemlinter of warning.scss should warn the use of an external block 1`] = `
"  ✓ warning
[warning.scss:21] Warning: \".external-block\" is only tolerated in this stylesheet.

OK: bemlinter has validated 1 block."
`;
